ateam_common/src/game_controller_listener.cpp

```diff
@@ -103,26 +103,86 @@ void GameControllerListener::RefereeMessageCallback(
 
   game_stage_ = static_cast<GameStage>(msg->stage);
 
-  // Yellow commands are even, blue are odd
-  // we_blue  odd(blue command)
-  // 0        0                   ours    leave it
-  // 0        1                   theirs  leave it
-  // 1        0                   theirs  add by 1 to be theirs
-  // 1        1                   ours    sub by 1 to be ours
-
-
-  // ignore if we dont know our team color yet
-  // only force start and above have this mapping
   uint8_t command = msg->command;
-  if (team_color_ != TeamColor::Unknown &&
-    command > static_cast<std::underlying_type_t<GameCommand>>(GameCommand::ForceStart))
-  {
-    if (team_color_ == TeamColor::Blue) {
-      bool command_is_blue = command % 2 == 1;
-      command += (command_is_blue ? -1 : 1);
-    }
+
+  // Cases in the switch here correspond with the options in the Referee message
+  // definition in the ssl_league_msgs package, translated to be for our/their
+  // team based on whether we are currently TeamColor::Yellow or TeamColor::Blue
+  // to the GameCommand format defined in the GameControllerListener's header file.
+  switch (command) {
+    case 0:
```

Comment:
We should use the named constants from [the message definition](https://github.com/SSL-A-Team/software/blob/main/ssl_league_msgs/msg/game_controller/Referee.msg)

For example:
```
    case ssl_league_msgs::msg::Referee::COMMAND_HALT:
```

Reply:
Done